cli/command_repository_set_parameters.go

```diff
@@ -140,32 +210,28 @@ func (c *commandRepositorySetParameters) run(ctx context.Context, rep repo.Direc
 		return errors.Wrap(err, "unable to get required features")
 	}
 
+	anyChange := false
 	upgradeToEpochManager := false
 
 	if c.upgradeRepositoryFormat {
-		anyChange = true
+		c.updateEpochParameters(&mp, &anyChange, &upgradeToEpochManager)
+	}
 
-		if !mp.EpochParameters.Enabled {
-			mp.EpochParameters = epoch.DefaultParameters()
-			upgradeToEpochManager = true
-			mp.IndexVersion = 2
-		}
+	c.setSizeMBParameter(ctx, c.maxPackSizeMB, "maximum pack size", &mp.MaxPackSize, &anyChange)
 
-		if mp.Version < format.FormatVersion2 {
-			mp.Version = format.FormatVersion2
+	// prevent downgrade of index format
+	if c.indexFormatVersion != 0 && c.indexFormatVersion != mp.IndexVersion {
```

Comment:
Might be a good idea to separate this change out in its own PR for better visibility.

cc @jkowalski , I believe the only scenario anyone would ever consider downgrading index version (outside of tests) is during a corrupted repository recovery attempts. Any concerns over disallowing this anymore ?